packages/cozy-client/src/CozyClient.js

```diff
@@ -187,6 +192,49 @@ class CozyClient {
     return this.mutate(this.getDocumentSavePlan(document), mutationOptions)
   }
 
+  ensureCozyMetadata(
+    document,
+    options = {
+      event: TRIGGER_CREATION
+    }
+  ) {
+    if (!document._type) return document
+
+    const schema = this.schema.getDoctypeSchema(document._type)
+    if (!schema.cozyMetadata) return document
+
+    const enriched = {
+      ...document
+    }
+    const metadata = Object.entries(schema.cozyMetadata)
+
+    for (const [
+      metadataName,
+      { trigger, value, useCurrentDate = false }
+    ] of metadata) {
+      const shouldSetMetadata =
+        options.event === TRIGGER_CREATION ||
+        (options.event === TRIGGER_UPDATE && trigger === TRIGGER_UPDATE)
+      if (!shouldSetMetadata) continue
+
+      let newValue
+
+      const shouldAccumulateValues = Array.isArray(value)
+      if (shouldAccumulateValues) {
+        const prevValue = get(document, `cozyMetadata.${metadataName}`, [])
+        newValue = uniqBy([...prevValue, ...value])
+      } else if (useCurrentDate) {
+        newValue = new Date().toISOString()
+      } else {
+        newValue = get(document, `cozyMetadata.${metadataName}`, value)
+      }
+
+      set(enriched, `cozyMetadata.${metadataName}`, newValue)
```

Comment:
Nit: since this string is used 3 times, you can store it in a `const path = `cozyMetadata.${metadataName}` for example.